addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -377,7 +377,9 @@ export class ProductScreen extends Component {
     }
 
     getProductListToNotDisplay() {
-        return [this.pos.config.tip_product_id, ...this.pos.pos_special_products_ids];
+        return this.pos.pos_special_products_ids.filter(
```

Comment:
I think you forget about the tip product which can be something else than a special product. Either add it here or, and I think it is better, change the method `_get_special_products` from `point_of_sale/models/pos_config.py` so that it also returns the tip product used by the config